File: src/custos/eval/harness/assistants/auto_approve.py
"""A1 ``auto_approve`` — unconditionally approves every denied call.

No LLM, no prompts, no policy synthesis.  baseline. Clean-room
re-implementation of the documented behaviour of
``Janus/src/permissions/assistants/auto_approve.py`` (the assistant whose
entire body is "return approve_once").

Does NOT persist a rule — Janus's ``auto_approve`` never emits ``create_policy``.
"""

from __future__ import annotations

from collections.abc import Mapping, Sequence
from typing import Any

from custos.eval.harness.assistants.base import BasePermissionAssistant
from custos.eval.harness.schema import JanusAssistantOutput, JanusAssistantVerdict

__all__ = ["AutoApproveAssistant"]


class AutoApproveAssistant(BasePermissionAssistant):
    name = "auto_approve"

    async def handle_permission_denial(
        self,
        subject: dict[str, Any],
        tool_name: str,
        action: str,
        args: dict[str, Any],
        failed_policies: Sequence[Mapping[str, Any]],
    ) -> JanusAssistantOutput:
        return JanusAssistantOutput(
            decision=JanusAssistantVerdict.APPROVE_ONCE,
            reason="auto-approve: unconditional one-time approval",
        )

File: packages/custos-ts/src/assistants/auto_approve.ts
// A1 — `auto-approve`. Mirrors `custos.assistants.rule_policy`-adjacent
// baseline (Janus A1). Unconditionally approves every denied call. No
// LLM, no prompts, no policy synthesis. Baseline.

import type { Assistant } from "./base.ts";
import type { AssistantOutput, Invocation, SubjectContext } from "../schema.ts";

export class AutoApproveAssistant implements Assistant {
  readonly name = "auto-approve";
  readonly exfiltratesArgs = false;

  decide(_inv: Invocation, _ctx: SubjectContext): AssistantOutput {
    return {
      decision: "allow_once",
      risk: 0,
      reasoning: "A1 auto-approve: unconditional baseline allow",
      fatigue_hint: false,
      persist_rule: null,
    };
  }
}